coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -596,11 +635,12 @@ func (server *Server) CompleteJob(ctx context.Context, completed *proto.Complete
 			return nil, xerrors.Errorf("get workspace build: %w", err)
 		}
 
+		workspace, getWorkspaceError := server.Database.GetWorkspaceByID(ctx, workspaceBuild.WorkspaceID)
```

Comment:
I think this `GetWorkspaceByID` needs to stay in the transaction, otherwise it's unsafe to update it. You could make it assign to a variable outside of the function to access the info!